ld_trials.js

```diff
@@ -0,0 +1,177 @@
+    choices: "NO_KEYS",
+    trial_duration: FIXCROSS_DURATION
+};
+
+/*
+ * This stimulus will present a forward mask
+ */
+let forward_mask = {
+    type: jsPsychHtmlKeyboardResponse,
+    stimulus: function(){
+        return "<p class='stimulus'>" +
+            jsPsych.timelineVariable('forward_mask', true) +
+            "</p>";
+    },
+    choices: "NO_KEYS",
+    trial_duration: MASK_DURATION,
+    data: { useful_data_flag: false }
+};
+
+/*
+ * This stimulus will present a visual prime stimulus
+ */
+let visual_prime = {
+    type: jsPsychHtmlKeyboardResponse,
+    stimulus: function(){
+        return "<p class='stimulus'>" +
+            jsPsych.timelineVariable('visual_prime', true) + "</p>";
+    },
+    choices: "NO_KEYS",
+    trial_duration: PRIME_DURATION,
+    post_trial_gap: PRIME_GAP_DURATION
+};
+
+/*
+ * This will present an auditory prime stimulus.
+ */
+let auditory_prime = {
+    type : jsPsychAudioKeyboardResponse,
+    stimulus : function() { return jsPsych.timelineVariable("auditory_prime");},
+    choices : "NO_KEYS",
+    trial_duration : PRIME_DURATION,
+    post_trial_gap : PRIME_GAP_DURATION
+}
+
+/*
+ * this will present a backwards mask.
+ */
+let backward_mask = {
+    type: jsPsychHtmlKeyboardResponse,
+    stimulus: function(){
+        return "<p class='stimulus'>" +
+            jsPsych.timelineVariable('backward_mask', true) +
+            "</p>";
+    },
+    choices: "NO_KEYS",
+    trial_duration: MASK_DURATION,
+    data: { useful_data_flag: false }
+}
+
+/**
+ * Returns an array that represents the allowed choices to respond.
+ *
+ * @return {(string|*)[]}
+ */
+function createChoicesArray() {
+    return [getWordKey(), getNonWordKey()];
+}
+
+/*
+ * This saves the data of a visual or auditory target stimulus
+ */
+function saveTargetData(data) {
+    let word_key = getWordKey();
+    let answer;
+    let correct;
+    let pressed_key = null;
+
+    if (data.response !== null) {
+        pressed_key = data.response.toUpperCase();
+    }
+
+    // Add "static" information to output
+    data.id = jsPsych.timelineVariable('id');
+    data.condition = jsPsych.timelineVariable('item_type');
+    data.word = jsPsych.timelineVariable('word');
+    data.expected_answer = jsPsych.timelineVariable('expected_answer');
+    data.forward_mask = jsPsych.timelineVariable('forward_mask');
+    data.visual_prime = jsPsych.timelineVariable('visual_prime');
+    data.auditory_prime = jsPsych.timelineVariable('auditory_prime');
+    data.backward_mask = jsPsych.timelineVariable('backward_mask');
+    data.auditory_target = jsPsych.timelineVariable('auditory_target');
+    data.visual_target = jsPsych.timelineVariable('visual_target');
+    data.useful_data_flag = true; // Mark this stimulus as import for analysis.
+
+    // compute correctness, pressed_key could be null in case of no response.
+    if (typeof pressed_key === 'string') {
+        answer = pressed_key === word_key ? 1 : 0;
```

Comment:
This is trivial, but I would rather set `data.answer` inside the if statement, instead of having an undefined `let answer` above, which is then used as a fallback below.

Reply:
I agree, this is a bit weird. Perhaps, I'm going to set answer to `null` above. 
when doing JSON.stringify() on the output keys with undefine values dissapear, but when the participant didn't respond it's nicer to have a 
```json
{
    "answer" : null
}
```
in the output than no answer key as well.

Reply:
Additionally, i'll remove the temporary `let correct` from the `saveTargetData()`, because I think it's fair to say that no answer, is an incorrect response.